_posts/2015-08-23-setting-up-spark-with-gradle.markdown

```diff
@@ -0,0 +1,62 @@
+---
+layout: post
+title: "Setting up Spark with Gradle"
+author: <a href="https://www.facebook.com/leonan.luppi" target="_blank">Leonan Luppi</a>
+date: 2015-08-23 10:24:03
+comments: true
+summary: >
+ We will learn how to set up a Spark Application using Gradle. For that we will use Eclipse Luna IDEA and Gradle 2.5.
+ --
+
+##Gradle
+ Gradle is a build automation tool build on top of concepts of Maven and Ant using groovy language instead of XML language. It can be use on Java, Groovy and Scala project, setting up tasks, dependencies and build order.
+
+##Installation
+First we need install gradle IDEA on Eclipse, so let's open our Eclipse IDEA:
+
+<br>Click in "Window" and select "Eclipse Marketplace..."
+<img src="/img/posts/settingup-spark-gradle/1.png" alt="">
+
+<br>Search for "gradle" term and select the first one: "Gradle IDE Pack XXX" then install normally and restart your Eclipse.
+<img src="/img/posts/settingup-spark-gradle/2.png" alt="">
+
+<br>Click in "New" and select "Other" then select "Gradle Project..."
+<img src="/img/posts/settingup-spark-gradle/3.png" alt="">
+
+<br>Put a "Project Name" then select Java Quickstart in "Sample Project"
+<img src="/img/posts/settingup-spark-gradle/4.png" alt="">
+
+<br>Look a familiar file struct, but instead of "pom.xml" now you have "build.gradle" let's open your "build.gradle"
+<img src="/img/posts/settingup-spark-gradle/5.png" alt="">
+
+<br>We have a "dependencies{}" object, we need to add spark-core dependencies on our project. **So add: compile 'com.sparkjava:spark-core:2.2'**
+<img src="/img/posts/settingup-spark-gradle/6.png" alt="">
+
+<br>We have a "dependencies{}" object, we need to add spark-core dependencies on our project.
+<pre><code class="language-markup">
```

Comment:
language-java (it's not 100% correct, but it gives okay syntax highlighting for this)